streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -150,6 +153,11 @@ void setMainConsumer(final Consumer<byte[], byte[]> mainConsumer) {
         this.mainConsumer = mainConsumer;
     }
 
+    /* For testing */
+    void setTaskIdToBackoffRecord(final Map<TaskId, BackoffRecord> taskIdToBackoffRecord) {
+        this.taskIdToBackoffRecord = taskIdToBackoffRecord;
+    }
```

Comment:
I do not think you need this method if you can control time as I describe on line 1017.